scripts/wgcna_analysis.R

```diff
@@ -0,0 +1,125 @@
+
+
+library(WGCNA);
+library(stringi);
+library(rjson)
+
+options(stringsAsFactors = FALSE);
+
+imgDir = Sys.getenv("GENERATED_IMAGE_DIR")
+# load expression data **assumes from json files row(traits)(columns info+samples)
+# pass the file_path as arg
+# pass the file path to read json data
+
+args = commandArgs(trailingOnly=TRUE)
+
+if (length(args)==0) {
+  stop("Argument for the file location is required", call.=FALSE)
+} else {
+  # default output file
+  json_file_path  = args[1]
+}
+
+inputData <- fromJSON(file = json_file_path)
+
+
+# parse the json data input
+
+minModuleSize <-inputData$minModuleSize
+
+TOMtype <-inputData$TOMtype
+
+corType <-inputData$corType
+# 
+
+trait_sample_data <- do.call(rbind, inputData$trait_sample_data)
+
+dataExpr <- data.frame(apply(trait_sample_data, 2, function(x) as.numeric(as.character(x))))
+# transform expressionData
+
+dataExpr <- data.frame(t(dataExpr))
+gsg = goodSamplesGenes(dataExpr, verbose = 3);
+
+# https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/
```

Comment:
**question:** Is this packaged anywhere?

Reply:
the  package being used in this script is wgcna already packaged in guix and the docs can be found here http://www.genetics.ucla.edu/labs/horvath/CoexpressionNetwork/Rpackages/WGCNA/. Removed the comment
above to avoid any confusion.